File: Java/InsertionSort.java
public class InsertionSort {

    public static void insertionSort(int[] array){

        for(int i = 1; i<array.length; i++){
            for(int j = 0; j<i; j++){
                if(array[i]<array[j]){
                    int temp = array[j];
                    array[j] = array[i];
                    array[i] = temp;
                }
            }
        }
    }

    public static void main(String[] args) {
        int[] array = {7, 5, 9, 3, 1, 10, 4, 2};
       insertionSort(array);
        for(int i = 0; i< array.length; i++){
            System.out.print(array[i] + " ");
        }

    }
}

File: Python/insertionSort.py
# Insertion sort

def insertionSort(array):

    for step in range(1, len(array)):
        key = array[step]
        j = step - 1
             
        while j >= 0 and key < array[j]:
            array[j + 1] = array[j]
            j = j - 1
        
        array[j + 1] = key

data = [9, 5, 1, 4, 3]
insertionSort(data)
print('Sorted Array:')
print(data)
